lib/glob-helpers.cjs

```diff
@@ -110,21 +110,26 @@ function normalizeFileForMatching(cwd, file) {
 
 exports.normalizeFileForMatching = normalizeFileForMatching;
 
+const normalizeNegatedPattern = (cwd, pattern) => {
+	// Remove `!` from pattern
+	const patternPath = pattern.slice(1);
+	const relativePath = path.relative(cwd, patternPath);
+	return `!${relativePath}`;
+};
+
 function normalizePattern(pattern) {
+	const cwd = process.cwd();
 	// Always use `/` in patterns, harmonizing matching across platforms
 	if (process.platform === 'win32') {
 		pattern = slash(pattern);
 	}
 
-	if (pattern.startsWith('./')) {
-		return pattern.slice(2);
-	}
-
-	if (pattern.startsWith('!./')) {
-		return `!${pattern.slice(3)}`;
+	if (pattern.startsWith('!')) {
+		return normalizeNegatedPattern(cwd, pattern);
```

Comment:
I don't think the helper function is all that useful, especially since it hides the critical `path.relative()` bit which is then called directly below.